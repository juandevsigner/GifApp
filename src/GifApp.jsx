import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifApp = () => {
  const [categories, setCategories] = useState([""]);

  const onNewCategory = category => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifApp</h1>
      <AddCategory onNewCategory={category => onNewCategory(category)} />
      {categories.map(category => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
