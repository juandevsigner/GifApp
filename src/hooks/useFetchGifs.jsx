import { useEffect, useState } from "react";
import { getGift } from "../helpers/getGifs";

export const useFetchGifs = category => {
  const [images, setImages] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getImages = async () => {
      const newImages = await getGift(category);
      setImages(newImages);
      setIsLoading(false);
    };
    getImages();
  }, []);

  return { images, isLoading };
};
