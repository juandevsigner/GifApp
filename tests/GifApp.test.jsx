import { fireEvent, render, screen } from "@testing-library/react";
import { GifApp } from "../src/GifApp";
import { useFetchGifs } from "../src/hooks/useFetchGifs";

describe("GifApp test", () => {
  it("vs snapshot", () => {
    const { container } = render(<GifApp />);
    expect(container).toMatchSnapshot();
  });
  it("should show tittle app", () => {
    render(<GifApp />);
    expect(screen.getByText("GifApp")).toBeTruthy();
  });
  it("should change input, reset input with submit and show category tittle", () => {
    render(<GifApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");
    fireEvent.input(input, { target: { value: "Marvel" } });
    expect(input.value).toBe("Marvel");
    fireEvent.submit(form);
    expect(input.value).toBe("");
    expect(screen.getByText("Marvel")).toBeTruthy();
  });
});
