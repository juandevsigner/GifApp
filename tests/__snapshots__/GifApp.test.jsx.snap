// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifApp test vs snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
